### ISSUE TITLE
UI issues

### ISSUE THREAD

**Issue by sharun-s** [kiwix/kiwix-js#267]
I have been spending more time in devtools than paying attention to the UI so just noticed a few things. Add to that on my local repo the article css wasn't loading (now fixed on my workers branch) plus some UI changes happened in between, so maybe these are older issues I just noticed. This is mostly on laptop screens (see attached image). They don't show up so much on mobile screen resolutions.
1. The "coordinates" div is somehow landing outside the article iframe
2. When you click a search result the articlelist and articleListHeaderMessage seem to be left behind occupying space. You can see the big gap below the searchbar and the article 
This is the temp fix I am using in handleTitleClick-
        $("#articleList").hide();
        $('#articleListHeaderMessage').hide(); 
3. The grey background on either side of the article, I guess is because article width is being fixed somewhere. Should the search bar be fixed width too? Right now there is this white space that extends to either end and looks a bit weird on big screens. Or probably the search bar can be moved into the nav bar above creating more space for the article? What do you think?   
Same issue with the transparent footer width at the bottom.
4. There is a small visible gap above the nav bar that seems to be produced by this rule
app.css [role=region] > header margin: .5rem 0 1rem 0; 
I have removed this locally. Don't know what it breaks though!
5. There are two scrollbars being produced. The page has one and the iframe has one (my screencap clipped the outer one). I have to click on the iframe to be able to scroll with the keyboard. Replacing the iframe with a div removes this issue (so only one scrollbar is visible) and scrolling feels natural with the keyboard.  The problem with using a div is applying the article css to the div within a style tag screws up the whole page. Maybe there is a simpler solution...   

![grab4](https://user-images.githubusercontent.com/18413274/27505166-f902170e-58b6-11e7-8684-3edd78e7f2b0.png)

**Comment by sharun-s** [comment]
And I just noticed the footer extends over both the scrollbars

**Comment by mossroy** [comment]
Thanks for this feedback.
We indeed did not invest much time on the UI, but had some ideas that might cover a few of your remarks, see #120 
Regarding the iframe, it clearly makes the UI more complicated, but we did not find a better solution. The need for an iframe (instead of a div) comes from the fact that we can not make any assumption on the content of a ZIM file. It can contain CSS and javascript that would conflict with the UI if it is not isolated inside an iframe (as you saw by yourself).

The UI relies on bootstrap and is in responsive design, so that it automatically adapts to small screens too.
Moving the search bar inside the nav bar seems to be a good idea IMHO, but we need to check that bootstrap supports that (else it would be a lot more work)

**Comment by sharun-s** [comment]
hmm...yup lots to think about...and do.

Not clear to me how that iframe gets loaded.
This line has me stumped $('#articleContent').contents().find('body').html($htmlArticle);
The iframe src which is the dummyArticle has its own head tag with some content. The code above seems to say, load the articlehtml with its own head (and css and js links) into the iframe body, yet somehow the dummyArticle's head disappears. I was thinking the dummy article head might allow us a way provide some global/custom css and js. Which might reduce the need for some of the css and js that gets loaded every article load. But it's looking like that's not how things work.

**Comment by sharun-s** [comment]
How do I add a label to filter for just UI issues?

**Comment by Jaifroid** [comment]
> I was thinking the dummy article head might allow us a way to provide some global/custom css and js. Which might reduce the need for some of the css and js that gets loaded every article load. But it's looking like that's not how things work.

Um,@sharun-s I already tried downloading the standard Wikipedia CSS and attaching it to the <head> area of dummyArticle:
```
    <head>
        <meta charset="utf-8">
        <title>dummy Article</title>
        <link rel="stylesheet" type="text/css" href="css/wikipedia-mobile.css"><!-- GK experimental: Added to start by using mobile stylesheet -->
    </head>
``` 
The result is that the mobile stylesheet is loaded at the start, and the article shows up with mobile stylesheet styling, but then the CSS gets replaced (probably simply overridden) in the final phase of loading the article, and uses the desktop stylesheet. This was just an experiment, but we can't use it anywhere seriously because the app isn't only for Wikipedia (and even so, different Wikipedia languages may have different stylesheets).

**Comment by sharun-s** [comment]
I feel it is doable. Would need some configuration setup and changes to the loading process. 
It was probably the idea behind having that dummyArticle there in the first place. I don't see a reason for it to be there otherwise.

**Comment by Jaifroid** [comment]
@sharun-s,  most of the UI stuff you mention is, I think, just part of the [Bootstrap](http://getbootstrap.com/) app development framework and library that's in this app. I've tried updating to Bootstrap 4 to see if it better supports Edge and irons out some of the display oddities (see #250) but it's too different from the 3.3.7 version which is the latest stable one that works. We might consider using a more responsive bootstrap theme, or using something like FramerJS instead (I'm interested in it because there are UWP themes published for FramerJS). But this would be longer term.

**Comment by sharun-s** [comment]
@Jaifroid Interesting and good to know. I should try out your win10 version sometime. Have some old Lumia phones lying around and have to say really liked their UI's.

**Comment by Jaifroid** [comment]
@sharun-s Ah, it depends how old those Lumias are. Many old models can't be updated to Windows 10 Mobile, at least not without rooting. Longer term I want to see if I can backport to Windows Phone 8.1, but its "Universal" app architecture and capabilities are different and it uses an older browser engine (not Edge).

If you run an Intel Windows 10 machine or VM, logged in with a Microsoft Account, you can try the UWP app on that platform, though it'll be easier to test when I've got a Beta in the Windows Store, coming soon....

**Comment by sharun-s** [comment]
Unfortunately looks like mine are 8.1, but I do have a win 10 machine I will try it on whenever your beta is ready!

**Comment by mossroy** [comment]
As far as I remember, this dummyArticle was introduced for the ServiceWorker mode, but I'm not 100% sure. If it's possible, I would really like to get rid of it (while keeping the ServiceWorker mode working, of course).
As @Jaifroid said, we can not presume of the ZIM content, so we can not apply a "generic" CSS.

**Comment by kelson42** [comment]
I think this quite old ticket should be updated and/or one ticket per problem - still there - should be open. Otherwise this ticket might be open for 5 additional years without much update.

**Comment by Jaifroid** [comment]
Yes @kelson42 , this can be closed because #523 is in progress and it will address some of these issues and change others. Closing.
